File: Prácticas1DAW/ExcepcionesPython/Ejercicio2.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Modifica la clase Gato vista anteriormente annadiendo el metodo apareaCon.
Este metodo debe comprobar que los gatos son de distinto sexo, tras lo cual, genera un nuevo gato.
Por ejemplo, seria valido algo como Gato cria = garfield.apareaCon(lisa). En caso de que los gatos sean del mismo sexo,
el metodo debe generar la excepcion ExcepcionApareamientoImposible. Crea un programa desde el que se pruebe el metodo.
"""

from excepciones.Gato import Gato
garfield = Gato("macho")
lisa = Gato("hembra")
felix = Gato("macho")

try:
    garfield.apareaCon(lisa)
    felix.apareaCon(garfield)
except:
    print("Excepcion capturada con exito")

File: Prácticas1DAW/ExcepcionesJava/Ejercicio2.java
package excepcionesJava;

public class Ejercicio2 {
	public static void main(String[] args) {
	
		Gato gato1 = new Gato("macho");
		Gato gato2 = new Gato("hembra");
		Gato gato3 = new Gato("macho");
		
		try {
			System.out.println(gato1.apareaCon(gato2));
			gato1.apareaCon(gato3);
		} catch (ApareamientoImposible excepcion) {
			System.err.println("Gatos del mismo sexo no se pueden aparear");
			//excepcion.printStackTrace();
		}
		
		
		
	}
}
